packages/xgboost/cmake/Toolchain.cmake

```diff
@@ -0,0 +1,26 @@
+set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS TRUE)
```

Comment:
Seems like it would be good to share this stuff with other CMake packages? Of course you would be better equipped to know about that than me. Perhaps as a followup to this PR or when you and @henryiii figure out more of the CMake stuff we could move this maybe to the `tools` directory and add a section to the package building docs about it?

For comparison, in #2378 I put the toolchain file into `tools`. It's not clear that this is the best place for these files, they might be better off in a folder called `toolchains` or something.

Reply:
> Perhaps as a followup to this PR or when you and @henryiii figure out more of the CMake stuff we could move this maybe to the `tools` directory and add a section to the package building docs about it?

Yeah, I think I should update our build system to use emscripten's cmake toolchain file directly if possible, instead of making a custom toolchain file for each package. I think it is possible if we set some compiler variables correctly but I haven't checked yet.